src/main/java/javaslang/collection/List.java

```diff
@@ -426,7 +426,15 @@ default int lastIndexOf(T element) {
     @Override
     default Tuple2<List<T>, List<T>> partition(Predicate<? super T> predicate) {
         Objects.requireNonNull(predicate, "predicate is null");
-        return Tuple.of(filter(predicate), filter(predicate.negate()));
+        List<T> left = nil(), right = nil();
+        for (T t : this) {
+            if (predicate.test(t)) {
+                left = left.prepend(t);
+            } else {
+                right = right.prepend(t);
+            }
+        }
+        return Tuple.of(left.reverse(), right.reverse());
```

Comment:
I'm pretty sure it doesn't harm, but don't we have a second pass here with the reverse() operation?